### PR TITLE
allow pydantic v2 with v1 api

### PR DESCRIPTION
**Pull request by loriab**
## Description
Apply Levi's trick like https://github.com/MolSSI/QCElemental/pull/314 and https://github.com/MolSSI/QCEngine/pull/412 to allow QCEngineRecords to work with pydantic v1 or v2

## Status
- [x] Ready to go

### CODE REVIEW

**Review by Lnaden — APPROVED** [review]

**Comment by Lnaden** [comment]
Does this have to have a release in order to work correctly in Engine, or is it just a data store effectively that Engine's tests pull from? I also don't have permission to approve this repo, so @bennybp will have to.

**Comment by loriab** [comment]
No release necessary, as qcng works off a qcnenginerecords commit stored in the former. I haven't found this a cureall, but because pip is involved, I'm not sure I'm doing it right either. I'll petition Ben for a merge.

**Review by bennybp — APPROVED** [review]

**Comment by loriab** [comment]
wait, it's wrong

File: qcenginerecords/info.py [hunk 1]
```diff
@@ -1,7 +1,10 @@
 import os
 import qcelemental as qcel
 from typing import Dict, List, Set
-from pydantic import BaseModel
+try:
+    from pydantic.v1.main import BaseModel
+except ImportError:
+    from pydantic.main import BaseModel
```
Comment (loriab) [inline qcenginerecords/info.py]:
```suggestion
try:
    from pydantic.v1 import BaseModel
except ImportError:
    from pydantic import BaseModel
```

**Comment by loriab** [comment]
ok, that's better. please go ahead.

**Pull request merged**